### PR TITLE
Support configuration by file

### PR DESCRIPTION
**Pull request by j16r**
Supports a chainlink.{toml,yaml,json} file for configuration, used by the deploy scripts and chainlink itself. There is now a `chainlink.toml` file in `internal/clroot` which sets common development settings that are not used by other components.

Note that configuration variables are uniformly defined using their ENV var representation to avoid confusion, e.g: "ROOT" not "RootDir".

[Finishes #155244740]

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -133,22 +133,22 @@ func Panicf(format string, values ...interface{}) {
 	logger.Panic(fmt.Sprintf(format, values...))
 }
 
-// Info logs an info message using Sprint.
+// Info logs an info message using Sprintf.
 func Info(args ...interface{}) {
```
Comment (dimroc) [inline logger/logger.go]: These methods actually aren't using `Sprintf`. That would mean we could pass tokens like `%v`, which isn't the case. There actually is an `Infof`, `Debugf`, later in the file that does indeed use `Sprintf`.

`Sprint` just takes a comma delimited list of parameters which it concatenates, while `Sprintf` leads with a format string, which is no in this function signature.

File: store/config.go [hunk 2]
```diff
@@ -312,6 +329,15 @@ func (c Config) SessionOptions() sessions.Options {
 	}
 }
 
+func (c Config) envVarName(field string) string {
+	schemaT := reflect.TypeOf(ConfigSchema{})
+	item, ok := schemaT.FieldByName(field)
```
Comment (dimroc) [inline store/config.go]: 👍

File: store/config.go [hunk 3]
```diff
@@ -114,40 +131,40 @@ func (c Config) BridgeResponseURL() *url.URL {
 
 // ChainID represents the chain ID to use for transactions.
 func (c Config) ChainID() uint64 {
-	return uint64(c.viper.GetInt64("ChainID"))
+	return uint64(c.viper.GetInt64(c.envVarName("ChainID")))
```
Comment (dimroc) [inline store/config.go]: A stretch goal that would be great would be to double down on reflection and see if we could pass the calling method's name to `c.envVarName(...)`, or do meta the go way™ and code generate into a separate file.

Comment (j16r) [reply store/config.go]: I did look into this, it's entirely possible - the runtime keeps the function call in the stack and you can inspect it. But thought it a bit too clever.

**Pull request merged**